test/integration/tcp_proxy_integration_test.cc

```diff
@@ -0,0 +1,125 @@
+#include "test/integration/tcp_proxy_integration_test.h"
+
+#include "common/network/utility.h"
+#include "common/ssl/context_manager_impl.h"
+
+#include "test/integration/ssl_utility.h"
+#include "test/integration/utility.h"
+#include "test/mocks/runtime/mocks.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
```

Comment:
If we're going to bikeshed on this, I'd love to see 
namespace Envoy {
namespace Test {
namespace {

where all the tests and tools are in the Envoy::Test namespace.   That'd definitely be out of scope for this PR so I'll do at most the anonymous namespace here :-)